.github/workflows/release.yml

```diff
@@ -27,13 +30,21 @@ jobs:
     - name: sign
       run: |
         # we smoke-test sigstore by installing each of the distributions
-        # we've built in a fresh environment and using each to sign for itself,
-        # using the ambient OIDC identity
+        # we've built in a fresh environment and using each to sign and
+        # verify for itself, using the ambient OIDC identity
         for dist in dist/*; do
           python -m venv smoketest-env
 
           ./smoketest-env/bin/python -m pip install "${dist}"
-          ./smoketest-env/bin/python -m sigstore sign "${dist}"
+
+          ./smoketest-env/bin/python -m \
+            sigstore sign --output-signature --output-certificate "${dist}"
+
+          ./smoketest-env/bin/python -m \
+            sigstore verify \
+            --cert "${dist}.crt" --signature "${dist}.sig" \
+            --cert-oidc-issuer https://token.actions.githubusercontent.com \
```

Comment:
IMO we need some way to verify the workflow (org, repo, name) here as well. Maybe not a blocker for this but something to think about

Reply:
Yep. These are the X.509v3 extensions we can verify against:

```
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature
            X509v3 Extended Key Usage: 
                Code Signing
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Key Identifier: 
                D2:A5:AA:A9:BF:8D:EA:35:7D:C5:6B:16:A8:45:68:A2:9A:15:2C:FB
            X509v3 Authority Key Identifier: 
                58:C0:1E:5F:91:45:A5:66:A9:7A:CC:90:A1:93:22:D0:2A:C5:C5:FA
            X509v3 Subject Alternative Name: critical
                URI:https://github.com/sigstore/sigstore-python/.github/workflows/release.yml@refs/tags/v0.4.2
            1.3.6.1.4.1.57264.1.1: 
                https://token.actions.githubusercontent.com
            1.3.6.1.4.1.57264.1.6: 
                refs/tags/v0.4.2
            1.3.6.1.4.1.57264.1.3: 
                dd2be1d19eeb0fbffbc36afe27bba7ba03fe97cf
            1.3.6.1.4.1.57264.1.5: 
                sigstore/sigstore-python
            1.3.6.1.4.1.57264.1.2: 
                release
            1.3.6.1.4.1.57264.1.4: 
                release
```

So we could additionally do the SAN here (since we have that via the misleading `--cert-email` option), as well as the commit hash and symbolic `ref`.

Reply:
Yes, I would agree checking SAN would make sense. My security property assumptions (that I have not yet checked against github docs) are 
* GitHub asserts the workflow referred to by SAN was used for this build
* GitHub asserts it only gives OIDC tokens to trusted GitHub build systems

This still leaves some details (like the choice of using _mutable_ tag in SAN) that could be improved by checking the other extensions if/when that is possible, but I think SAN+OIDC provider is a very good starting point.